modules/core/src/main/java/org/apache/ignite/metric/IgniteMetricRegistry.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.metric;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+import java.util.function.IntSupplier;
+import java.util.function.LongSupplier;
+import java.util.function.Supplier;
+import org.apache.ignite.lang.IgniteExperimental;
+import org.apache.ignite.spi.metric.BooleanMetric;
+import org.apache.ignite.spi.metric.DoubleMetric;
+import org.apache.ignite.spi.metric.IntMetric;
+import org.apache.ignite.spi.metric.LongMetric;
+import org.apache.ignite.spi.metric.ObjectMetric;
+import org.apache.ignite.spi.metric.ReadOnlyMetricRegistry;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Metric registry. Allows to get, add or remove metrics.
+ *
+ * @see IgniteMetrics
+ * @see ReadOnlyMetricRegistry
+ */
+@IgniteExperimental
+public interface IgniteMetricRegistry extends ReadOnlyMetricRegistry {
+    /**
+     * Registers an int metric which value will be queried from the specified supplier.
+     *
+     * @param name Metric name.
+     * @param supplier Metric value supplier.
+     * @param desc Metric description.
+     * @return New or previously registered {@link IntMetric}.
+     */
+    IntMetric register(String name, IntSupplier supplier, @Nullable String desc);
```

Comment:
Do we really need to return `IntMetric` here and in other methos?
After registration metrics exposed by exporter.
What are scenarios when user need to store `IntMetric` instance?

Reply:
It is not necessary but looks handy. User may read metric in its own way, not only by an exporter. We do similar, for instance, with `CacheMetricImpl#offHeapEntriesCnt, offHeapPrimaryEntriesCnt, heapEntriesCnt, cacheSize`. Using same API. It is simple to return additional getters.

Reply:
User can re-use supplier to read metric value. There is no need to return `IntMetric` here and in other methods.
Let's return `void` here.